rust/cache/src/foyer.rs

```diff
@@ -465,9 +465,10 @@ where
 
     async fn insert(&self, key: K, value: V) {
         let _stopwatch = Stopwatch::new(&self.insert_latency);
-        self.cache.insert(key.clone(), value.clone());
-        // Also insert to the disk cache.
-        self.insert_to_disk(key, value);
+        // Immediately insert to the disk cache
+        self.insert_to_disk(key.clone(), value.clone());
```

Comment:
the order of operations here is important: foyer will remove the entry from the in-memory cache (if populated) when inserting to disk, so we should insert to disk first to avoid unnecessary disk reads on every first `.get()` call for a key